src/components/DpAutocomplete/DpAutocomplete.vue

```diff
@@ -1,114 +1,396 @@
 <template>
-  <vue-omnibox
-    :box-height="height"
-    :disable-search="true"
-    :label="label"
-    :options="options"
-    :placeholder="placeholder"
-    :value="currentQuery"
-    @input="emitAndFetch"
-    @searched="emit($event, 'searched')"
-    @selected="emit($event, 'selected')">
-    <template
-      v-for="(_, slotName) in $slots"
-      v-slot:[slotName]="slotData">
-      <slot
-        v-bind="slotData"
-        :name="slotName" />
-    </template>
-  </vue-omnibox>
+  <div class="relative">
+    <!-- Search container -->
+    <div
+      :class="[
+        isInputFocused ? 'border-interactive' : 'border-gray-300',
+        { 'focus-within:ring-1 focus-within:ring-interactive': isInputFocused }
+      ]"
+      class="border rounded-md flex items-center overflow-hidden relative transition-colors duration-200 ease-in-out"
+      role="combobox"
+      aria-haspopup="listbox"
+      aria-controls="options-list"
+      :aria-expanded="isOptionsListVisible"
+      tabindex="0"
+      @mousedown.stop.prevent="focusInput">
+      <label
+        id="autocomplete-label"
+        for="input-box"
+        :class="{
+          'h-full w-full px-1 py-[2px] text-gray-400 flex items-center m-0': true,
+          'absolute pointer-events-none': !isPlaceholderVisible,
+          'sr-only': !isPlaceholderVisible
+        }">
+        {{ placeholder }}
+      </label>
+      <div
+        id="input-box"
+        ref="input"
+        role="textbox"
+        :class="{
+          'sr-only': isPlaceholderVisible,
+          'w-auto min-w-[10px]': isInputFocused,
+          'w-full': !isInputFocused
+        }"
+        class="py-[2px] px-1 outline-none whitespace-pre overflow-hidden"
+        contenteditable="true"
+        tabindex="0"
+        aria-labelledby="autocomplete-label"
+        :style="boxHeightStyle"
+        @mousedown.stop.prevent="focusInput"
+        @keydown.stop="runSpecialKeys"
+        @focusout="isInputFocused = false" />
+      <div
+        v-show="isInputFocused"
+        :style="boxHeightStyle"
+        class="w-auto max-w-full py-[2px] text-gray-300 overflow-hidden whitespace-pre flex items-center"
+        aria-hidden="true">
+        {{ completion }}
+      </div>
+    </div>
+
+    <!-- Options list -->
+    <div class="relative">
+      <ol
```

Comment:
sugesstions list should have a higher z-index to prevent this issue or the list below should have a lower index
![image](https://github.com/user-attachments/assets/23594fc8-1fef-4ef6-92d4-772139d3b0b3)

Reply:
`z-flyout` would do, i guess.